spec/requests/telegram_webhooks_spec.rb

```diff
@@ -0,0 +1,172 @@
+      end
+    end
+
+    context 'when not logged' do
+      before do
+        dispatch_command :start
+      end
+
+      it { is_expected.to respond_with_message 'Hi! Give me your mobile number.' }
+
+      it 'present phone number key' do
+        expect(bot.requests[:sendMessage].last[:reply_markup]).to eq keyboard: [
+          [{ text: 'Send Phone Number', request_contact: true }]
+        ],
+                                                                     one_time_keyboard: true,
+                                                                     resize_keyboard: true
+      end
+    end
+  end
+
+  describe '#auth' do
+    context 'when user has account on website' do
+      subject { -> { dispatch_message '', contact } }
+
+      before { dispatch_command :start }
+
+      it { is_expected.to respond_with_message('Choose action') }
+    end
+
+    context 'when user hasn\'t account on website' do
+      subject { -> { dispatch_message '', { contact: { phone_number: '1112223334' } } } }
+
+      before { dispatch_command :start }
+
+      it { is_expected.to respond_with_message('User not found.') }
+    end
+  end
+
+  describe '#help!' do
+    subject { -> { dispatch_command :help } }
+
+    it { is_expected.to respond_with_message(/Available commands:/) }
+  end
+
+  describe '#companies!' do
+    subject { -> { dispatch_command :companies } }
+
+    context 'when logged' do
+      before do
+        dispatch_command :start
+        dispatch_message '', contact
+      end
+
+      it { is_expected.to respond_with_message(user.companies.map(&:name).join(', ')) }
+    end
+
+    context 'when not logged' do
+      it { is_expected.to respond_with_message('Hi! Give me your mobile number.') }
+    end
+  end
+
+  describe '#add_meter_reading!' do
+    subject { -> { dispatch_command :add_meter_reading } }
+
+    context 'when logged' do
+      before do
+        dispatch_command :start
+        dispatch_message '', contact
+      end
+
+      it { is_expected.to respond_with_message('Choose utility provider') }
+      it { expect(bot.requests[:sendMessage].last[:reply_markup]).to be_present }
+    end
+
+    context 'when not logged' do
+      it { is_expected.to respond_with_message('Hi! Give me your mobile number.') }
+    end
+  end
+
+  describe '#add_meter' do
+    subject { -> { dispatch_message utility_provider.company.name } }
+
+    before do
+      dispatch_command :start
+      dispatch_message '', contact
+      dispatch_command :add_meter_reading
+    end
+
+    it { is_expected.to respond_with_message("Write meter reading") }
+  end
+
+  describe '#save_meter_reading' do
+    before do
+      dispatch_command :start
+      dispatch_message '', contact
+      dispatch_command :add_meter_reading
+      dispatch_message utility_provider.company.name
+    end
+
+    context 'when correct meter reading' do
+      subject { -> { dispatch_message meter_reading } }
```

Comment:
why not just subject { dispatch_message }

Reply:
dispach_message must has minimum one argument
![Screenshot from 2020-11-06 10-44-39](https://user-images.githubusercontent.com/31774981/98345307-29a8c980-201d-11eb-82cb-5866f7538fea.png)